jujugui/static/gui/src/app/app.js

```diff
@@ -1931,12 +1952,31 @@ YUI.add('juju-gui', function(Y) {
       if (environmentInstance) {
         environmentInstance.topo.update();
       }
+      this.set('modelUUID', '');
       this.set('loggedIn', false);
-      this.env.logout();
-      this.controllerAPI.logout();
-      this.maskVisibility(true);
-      this._renderLogin(null);
-      return;
+      // Close both controller and model API connections.
+      let closeController = callback => {
+        callback();
+      };
+      const controllerAPI = this.controllerAPI;
+      if (controllerAPI) {
+        closeController = controllerAPI.close.bind(controllerAPI);
+      }
+      this.env.close(() => {
+        closeController(() => {
+          if (controllerAPI) {
+            // Juju 2 connects to the controller and gets models from there.
+            controllerAPI.connect();
```

Comment:
Yes, because that's the only way the GUI can make it, the GUI is nothing without a controller connection, being it Juju, Jimm or the fake backend. Maybe you are thinking about authenticating? We are not logging in again, we are just reconnnecting the WebSocket here.